### PR TITLE
SDIT-1226 initial court case mappings

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourtCase.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  val statusUpdateReason: String? = null,
+
+  val statusUpdateDate: LocalDate? = null,
+
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "STATUS_UPDATE_STAFF_ID")
+  val statusUpdateStaff: Staff? = null,
+
+  // always has a value
+  val lidsCaseNumber: Int = 1,
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourtCase.kt:98]: no nulls on database, not sure if this actually is a case number - can discuss

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  val combinedCaseId: Long?,
+  val statusUpdateStaffId: Long?,
+  val statusUpdateDate: LocalDate?,
+  val statusUpdateComment: String?,
+  val statusUpdateReason: String?,
+  val lidsCaseId: Int?,
+  val lidsCombinedCaseId: Int?,
+  val lidsCaseNumber: Int,
+  val createdDateTime: LocalDateTime,
+  val createdByUsername: String,
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt:104]: the majority of field comments don't add value here - tempted to leave off for this internal api.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.toCodeDescription
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Offender
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderBooking
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.CourtCaseRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.findRootByNomisId
+
+@Service
+@Transactional
+class SentencingService(
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt:18]: will revisit naming as domain becomes clearer

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  private val courtCaseRepository: CourtCaseRepository,
+  private val offenderRepository: OffenderRepository,
+  private val telemetryClient: TelemetryClient,
+) {
+  fun getCourtCase(id: Long, offenderNo: String): CourtCaseResponse {
+    findPrisoner(offenderNo).findLatestBooking()
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt:24]: this validation of offender/booking may be a. bit overboard here - especially if used in migration.  Leaving in for now as may assist with development/testing

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourtCase.kt [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  val caseInfoNumber: String? = null,
+
+  @OneToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "COMBINED_CASE_ID")
+  val combinedCase: CourtCase?,
+
+  val statusUpdateComment: String? = null,
+
+  // no matching reference domain
+  val statusUpdateReason: String? = null,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourtCase.kt:89]: any idea how they input this into NOMIS?

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**